src/xtgeo/surface/regular_surface.py

```diff
@@ -2069,14 +2070,27 @@ def smooth(self, method="median", iterations=1, width=1):
         """Various smoothing methods for surfaces.
 
         Args:
-            method: Smoothing method (median)
+            method: Smoothing method (median or gaussian)
             iterations: Number of iterations
             width: Range of influence (in nodes)
 
         .. versionadded:: 2.1
         """
+
         if method == "median":
-            _regsurf_gridding.smooth_median(self, iterations=iterations, width=width)
+            _regsurf_gridding._smooth(
+                self,
+                window_function=scipy.ndimage.median_filter,
+                iterations=iterations,
+                width=width,
+            )
+        elif method == "gaussian":
+            _regsurf_gridding._smooth(
+                self,
+                window_function=scipy.ndimage.gaussian_filter,
+                iterations=iterations,
+                width=width,
+            )
         else:
             raise ValueError("Unsupported method for smoothing")
```

Comment:
It may be possible to compress this a bit since the call to `_smooth` has same pattern, aka:

```
window_function = None
if method in ["median", "gaussian"]:
     window_function = global()[f"scipy.ndimage.{method}_filter"]   # https://pythonhow.com/how/call-a-function-of-a-module-by-using-its-name/
else:
    raise Valuerror(...)     

_regsurf_gridding._smooth(
    self,
    window_function=window_function,
    iterations=iterations,
    width=width,
)
```

Reply:
Neat, but not a fan. Its a bit hard to read, having references to the functions makes them _clickabole_ by most editors so you get the docs... without having to fight the code.

Reply:
That is a point; but at least can reduce to one call for `_smooth()` and let `if...` steer the `scipy` method

Reply:
It seems like a premature optimization. The additional overhead from an extra function call is negligible compared to the runtime of the filter functions in this context.